### PR TITLE
Send an error report to slack - not parsing the output yet

### PR DESCRIPTION
**Pull request by SvenDowideit**
Signed-off-by: Sven Dowideit SvenDowideit@home.org.au

### CODE REVIEW

File: vars/documentationChecker.groovy [hunk 1]
```diff
@@ -15,39 +15,41 @@ def call(String docsDir) {
...
+        sh "docker build -t ${imageName} ${docsDir}"
+        try {
+          sh "docker run --name=${containerName} ${imageName}"
+
+          // TODO: summarize the changes & errors (these are files used by GHPRB and the summary plugin
+          sh "docker cp ${containerName}:/validate.junit.xml ."
+          sh "docker cp ${containerName}:/docs/markdownlint.summary.txt ."
+
+          //setGitHubPullRequestStatus message: "docs test complete"
+        } finally {
```
Comment (SvenDowideit) [inline vars/documentationChecker.groovy]: mmm, i think i need a 

```
 catch (err) { throw err } here..

**Comment by SvenDowideit** [comment]
@mikedougherty this is getting long, and messy

I'm wondering if I can define more methods in the var, but still keep the `call()` method too:

```
// The method most users will call using documentationChecker("docs")
def call() {
  try {
    this.innerChecker()
  } catch (err) {
    slackSend...
  }
}
def setDebug(v) {
    this.debug = v;
}
def getDebug() {
    return this.debug;
}
def innerChecker() {
  ...
}
```

**Comment by mikedougherty** [comment]
according to this, you can: https://github.com/jenkinsci/workflow-cps-global-lib-plugin#defining-global-variables

**Comment by SvenDowideit** [comment]
I must be blind - I see nothing about whether that allows `call()` to still work. I'll try it next time i poke this code - if this version is good enough, please mergify :)

**Pull request merged**